frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java

```diff
@@ -208,9 +214,12 @@ private boolean handleDivergedFromAndNewerThanRemote(IManagedBranchSnapshot gitM
 
     switch (forcePushApprovalDialogBuilder.show(project)) {
       case YES :
-        // TODO (#1424): git push continues in background
-        new GitPushDialog(project, gitRepository, GitPushSource.create(localBranch), /* isForcePushRequired */ true).show();
-        return true;
+        Runnable doInUIThreadWhenReady = () -> graphTable.queueRepositoryUpdateAndModelRefresh(traverseNextEntry);
+        new GitPushDialog(project, gitRepository, GitPushSource.create(localBranch), /* isForcePushRequired */ true,
+            doInUIThreadWhenReady).show();
+        // The ongoing traverse is now a responsibility of the freshly-queued backgroundable;
```

Comment:
copy paste error detected 🔴  `backgroundable` - not precise

Reply:
oh I see the next PR is removing these comments so feel free to ignore me on this one